SSSOM.md

```diff
@@ -491,3 +492,77 @@ Not part of spec, clean up.
 * Can we have more complex logical mappings? A common use case is a taxon-specific anatomy ontology (taxon X). We want to map A to B, but say that B is not equivalent to A, rather it is equivalent to an OWL expression based on A (A and 'in taxon' some X). But there are plenty of arbitrary expressions that could be supported. A while back I looked at some SNOMED to Uberon mappings; it seemed like a lot of the SNOMED terms would better be ('part of' some UBERON:X) rather than directly equivalent to X.
 *  explicitly declare the metadata elements as equivalent to properties?
 * What about mapping to Literals in general? Is this in-spec?
+
+<a name="minimum"></a>
+
+### Minimum Metadata Recommendation
+
+Current mappings are extremely hard to use for data integration, because they are:
+
+- non-transparently imprecise
+- non-transparently incomplete
+- inaccurate
+- unFAIR
+
+In principle, to reach full integration through mappings, you will have to cross-map all ontologies, or semantic
+spaces, which means if you have N "spaces", you have N x N (-1) mappings. To mitigate the explosion
+of mappings, we have to be able to cross-walk:
+- multi-hop forward walks `{ O1:A->O2:A, O2:A->O3:A } --> {O1:A->O3:A}`
+- walk-backs `{ O1:A->O2:A } --> {O2:A->O1:A}`
+- combinations `{ O1:A->O2:A, O2:A->O3:A, O4:A->O3:A } --> {O1:A->O4:A}`
+
+To enable cross-walking, we propose the following 5 star system for mapping metadata. 
+
+1. _1-star mappings_ fulfill the following criteria:
+   - record subject id, object id and mapping precision
+   - using qualified names (either URIs or CURIEs + curie maps) for subject id and object id
+   - using a standard file format (JSON, XML, CSV, TSV)
+   - made available in a public space
+   - _optional_: record the subject and object labels to make it easier for humans to read the file
+2. _2-star mappings_ fulfill all the criteria for 1-star mappings and furthermore
+   - made available in a public version control system with an issue tracker
+   - record the semantic predicate explicitly
+   - using qualified names for the semantic predicate (i.e. owl:equivalentClass, skos:exactMatch)
+   - record a confidence value for the mapping between 0 and 1.
+   - record an open license for the use of the mapping set
+3. _3-star mappings_ fulfill all the criteria for 2-star mappings and furthermore 
+   - are stored in SSSOM format
+   - record the following additional metadata: 
+     - `match_type`(s) (Lexical, Logical match, HumanCurated etc)
+     - `date` of the mapping
+     - `creator_id`
+     - `subject_source`
+     - `object_source`
+     - `subject_source_version`
+     - `object_source_version`
+     - `mapping_tool` if the mapping was automatically computed using a tool
+4. _4-star mappings_ fulfill all the criteria for 3-star mappings and furthermore
+   - Register the mapping at a mapping commons
+   - record the following additional metadata:
+     - `mapping_set_id`
```

Comment:
maybe mention that this ID should be unique in combination with the `mapping_provider`

Reply:
I think we should actually propose this to be globally unique, using a w3id purl. I would really like that!

Reply:
Agree for global uniqueness. what would such a PURL point to? Should people have to register these somewhere? If we were to provide a way to register, maybe we could end-to-end automate it via github issues like I did in the bioregistry and also have a website get autogenerated

Reply:
I could set that up in a different repo in the mapping-commons namespace if the idea makes sense

Reply:
Yes, I already have some basic idea how this could work; the only piece missing is workflow: basically my idea is that if you run a `mapping-commons`,  you basically make a registration request on a central mapping-commons repo, or you go straight to https://github.com/perma-id/w3id.org/blob/master/sssom/.htaccess to make a PR. 

It is a bit risky; I am wondering whether w3id would accept file protection on a user basis. Alternatively, is there a way to manage rewrite rules (say for  `http://w3id.org/sssom/mappings/my-commons/my-mapping.sssom.tsv`)  somehow outside htaccess and import them? No idea.